packages/toolpad-core/src/AppProvider/AppProvider.tsx

```diff
@@ -110,6 +121,18 @@ function AppProvider(props: AppProviderProps) {
     window: appWindow,
   } = props;
 
+  React.useEffect(() => {
```

Comment:
To avoid an unnecessary effect, how about the following: During render, for each rendered navigation item, calculate the pathname, then look it up in a `Set`, if it already exists in the set, warn and don't mark it active. Add the pathname to the set.